### PR TITLE
Updates to documentation

### PR DESCRIPTION
**Pull request by dosaboy**
Extend variables docs and reorganise properties. Requires property is now part of "main" properties since it can be used on its own.

### CODE REVIEW

File: doc/source/contrib/language_ref/property_ref/main_properties.rst [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+Settings
+^^^^^^^^
+
+decision
+""""""""
+
+This property is typically used in :ref:`conclusions`.
+CHECKS refers to a set of one or more :ref:`checks` names organised as a
+:ref:`LogicalCollection` to make decision on the outcome of more
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/main_properties.rst:93]: s/to make a decision on the outcome of one or more

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    CACHE_KEYS
+      package
+      revision
+
+pebble
+------
+
+Takes a pebble service and optionally some parameters to check.
+Returns True if service exists and, if provided, parameters match.
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/requirement_types.rst:106]: s/if the service

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+      ...
+
+    CACHE_KEYS
+      services
+
+systemd
+-------
+
+Takes a systemd service and optionally some parameters to check.
+Returns True if service exists and, if provided, parameters match.
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/requirement_types.rst:143]: s/if the service

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+      ...
+
+    CACHE_KEYS
+      services
+
+config
+------
+
+A dictionary containing the information required to perform some config checks.
+Supports applying assertion rules to the contents of one or more config file.
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/requirement_types.rst:187]: s/one or more config files

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+.. code-block:: python
+
+    input.paths
+
+search
+------
+
+Used to define search expression(s) and criteria. There are different types of
+search expression that can be used depending on the data being searched and how
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/shared_properties.rst:70]: s/search expressions

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+the results will be interpreted:
+
+A "simple" search (`SearchDef <https://github.com/dosaboy/searchkit/tree/main/searchkit>`_) involves a single pattern
+and is used to match single lines.
+
+A "sequence" search (`SequenceSearchDef <https://github.com/dosaboy/searchkit/tree/main/searchkit>`_) to match
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/shared_properties.rst:76]: s/is used to match (at end of line)

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+`LogEventStats <https://github.com/canonical/hotsos/tree/main/hotsos/core/analytics.py>`_. This requires a
+callback method to be implemented to process the results and is designated using
+the optional *passthrough-results* option. Search results are passed
+to their handler as a raw `SearchResultsCollection <https://github.com/dosaboy/searchkit/tree/main/searchkit>`_.
+
+NOTE: this property is implemented as a :ref:`mapped property <mappedproperties>` so the *search* name is optional.
+
+IMPORTANT: do not use global search properties. If you do this, the same search
+tag will be used for all searches and it will not be possible to
+distinguish results form more than one leaf node.
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/shared_properties.rst:89]: s/form/from

File: doc/source/contrib/language_ref/property_ref/variables.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    bfoo: true
+
+Or they can reference a Python property. This is done by prefixing the import string with '@' e.g.
+
+.. code-block:: yaml
+
+  vars:
+    foo: '@path.to.myproperty'
+
+When *$foo* used it will have the value of *myproperty*.
```
Comment (brianphaley) [inline doc/source/contrib/language_ref/property_ref/variables.rst:22]: s/is used

**Review by brianphaley — CHANGES_REQUESTED** [review]

File: doc/source/contrib/language_ref/property_ref/main_properties.rst (hunk 1 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/main_properties.rst:93]: ack.

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst (hunk 2 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/requirement_types.rst:106]: ack

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst (hunk 3 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/requirement_types.rst:143]: ack

File: doc/source/contrib/language_ref/property_ref/requirement_types.rst (hunk 4 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/requirement_types.rst:187]: ack

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst (hunk 5 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/shared_properties.rst:70]: ack

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst (hunk 6 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/shared_properties.rst:76]: ack

File: doc/source/contrib/language_ref/property_ref/shared_properties.rst (hunk 7 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/shared_properties.rst:89]: ack

File: doc/source/contrib/language_ref/property_ref/variables.rst (hunk 8 above)
Comment (dosaboy) [reply doc/source/contrib/language_ref/property_ref/variables.rst:22]: ack

**Pull request merged**